server/controllers/controllers.js

```diff
@@ -0,0 +1,31 @@
+import mongoose from 'mongoose'
+import Recipe from '../models/recipe'
+
+
+exports.root = (req, res) => {
+    res.render('index')
+}
+
+exports.publish = (req, res) => {
+    res.render('post')
+}
+
+
+exports.allRecipes = async (req, res) => {
+  const recipes = await Recipe.find()
+  res.json(recipes)
+}
+
+exports.recipe = async (req, res) => {
+  const slug =  req.params.recipe
+  const recipe = await Recipe.find({slug: slug})
+
+  res.send(recipe)
+}
+
+exports.postRecipe = async (req,res) => {
+  const recipe = new Recipe(req.body)
+    await recipe.save()
+    req.addFlashMessage('success', 'thanks!, the recipe has been published successfuly')
+    res.redirect(`/recipes/${recipe.slug}`)
+}
```

Comment:
actually i am handling the errors when calling the async function by attach the catch method to ti, instead of using catch try block like this: 

+  controllers.postRecipe(req, res).catch((err) => {
+    req.addFlashMessage('error', 'Oops some went wrong! please try again and make sure to fill all fields')